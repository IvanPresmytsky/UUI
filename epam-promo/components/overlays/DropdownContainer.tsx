import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';

type DropdownContainerMods = {
    /**
     * Defines component color.
     * @default 'white'
     * */
    color?: 'white' | 'gray70';
};

/** Represents the properties of a DropdownContainer component. */
export type DropdownContainerProps = uui.DropdownContainerProps & DropdownContainerMods;

function applyDropdownContainerMods(mods: DropdownContainerMods) {
    return [`uui-color-${mods.color || 'white'}`];
}

export const DropdownContainer = createSkinComponent<uui.DropdownContainerProps, DropdownContainerProps>(
    uui.DropdownContainer,
    () => null,
    applyDropdownContainerMods,
);
