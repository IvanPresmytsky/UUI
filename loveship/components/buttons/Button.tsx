import { FillStyle, ControlShape, EpamPrimaryColor } from '../types';
import * as uui from '@epam/uui';
import { createSkinComponent, devLogger } from '@epam/uui-core';
import { systemIcons } from '../icons/icons';
import css from './Button.module.scss';

const defaultSize = '36';

export type ButtonColorType = EpamPrimaryColor | 'white' | 'night500' | 'night600' | 'gray';

export interface ButtonMods {
    /** @default "sky"  */
    color?: ButtonColorType;
    /** @default '36' */
    size?: uui.ControlSize | '18';
    /** @default 'square' */
    shape?: ControlShape;
    /** @default 'solid' */
    fill?: FillStyle;
}

const mapFill: Record<FillStyle, uui.ButtonFill> = {
    solid: 'solid',
    white: 'outline',
    light: 'ghost',
    none: 'none',
};

export type ButtonProps = uui.ButtonCoreProps & ButtonMods;

export function applyButtonMods(mods: ButtonProps) {
    return [
        `uui-size-${mods.size || defaultSize}`,
        css['style-' + (mods.shape || 'square')],
    ];
}

export const Button = createSkinComponent<uui.ButtonProps, ButtonProps>(
    uui.Button as any, // TODO: remove it when BaseButton inheritance will be reworked
    (props) => {
        if (__DEV__) {
            devLogger.warnAboutDeprecatedPropValue<ButtonProps, 'color'>({
                component: 'Button',
                propName: 'color',
                propValue: props.color,
                propValueUseInstead: 'gray',
                condition: () => ['night500', 'night600'].indexOf(props.color) !== -1,
            });
        }
        return {
            dropdownIcon: systemIcons[props.size || defaultSize].foldingArrow,
            clearIcon: systemIcons[props.size || defaultSize].clear,
            fill: mapFill[props.fill] || mapFill.solid as any,
        };
    },
    applyButtonMods,
);
