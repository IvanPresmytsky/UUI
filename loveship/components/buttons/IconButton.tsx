import * as uui from '@epam/uui';
import { createSkinComponent, devLogger } from '@epam/uui-core';
import { EpamAdditionalColor, EpamPrimaryColor } from '../types';

type IconButtonMods = {
    /**
     * Defines component color.
     * @default 'night600'
     */
    color?: EpamPrimaryColor | EpamAdditionalColor | 'white' | 'night200' | 'night300' | 'night400' | 'night500' | 'night600';
};

/** Represents the properties of a IconButton component. */
export type IconButtonProps = uui.IconButtonCoreProps & IconButtonMods;

export const IconButton = createSkinComponent<uui.IconButtonProps, IconButtonProps>(
    uui.IconButton,
    (props) => {
        if (__DEV__) {
            devLogger.warnAboutDeprecatedPropValue<IconButtonProps, 'color'>({
                component: 'IconButton',
                propName: 'color',
                propValue: props.color,
                condition: () => ['night200', 'night300', 'night400'].indexOf(props.color) !== -1,
            });
        }
        return {
            color: props.color ?? 'night600',
        };
    },
);
