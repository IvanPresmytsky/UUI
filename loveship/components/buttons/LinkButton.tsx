import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';

type LinkButtonColor = 'sky' | 'grass' | 'fire' | 'night100' | 'night600' | uui.LinkButtonProps['color'];

type LinkButtonMods = {
    /**
     * Defines component color.
     * @default 'sky'
     */
    color?: LinkButtonColor;
};
// TODO: leave 3 colors white, blue, gray - more specific colors will be determined later.
/** Represents the properties of a LinkButton component. */
export type LinkButtonProps = uui.LinkButtonCoreProps & LinkButtonMods;

export const LinkButton = createSkinComponent<uui.LinkButtonProps, LinkButtonProps>(
    uui.LinkButton,
    (props) => {
        return {
            color: props.color ?? 'sky',
        };
    },
    () => [],
);
