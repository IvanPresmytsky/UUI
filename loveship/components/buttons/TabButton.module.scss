$text-size-caption: 36px;
$text-size-count: 24px;

@mixin tab-button-color($bg, $fg, $brd) {
    color: $fg;
    fill: $fg;

    &:hover, &:focus {
        fill: $bg;
        color: $bg;
    }

    &:global(.uui-active) {
        fill: $bg;
        color: $bg;
        border-color: $brd;
    }
}

.theme-dark {
    @include tab-button-color(var(--sky-50), var(--night300), var(--sky-50));

    &:global(.uui-disabled) {
        @include tab-button-color(var(--gray60), var(--gray60), var(--gray60));
    }
}
