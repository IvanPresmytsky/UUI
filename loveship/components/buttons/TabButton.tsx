import css from './TabButton.module.scss';
import * as uui from '@epam/uui';
import { withMods } from '@epam/uui-core';

export type TabButtonMods = uui.TabButtonProps & {
    theme?: 'light' | 'dark';
};

function applyTabButtonMods(mods: TabButtonMods & uui.TabButtonProps) {
    return [mods.theme === 'dark' && css.themeDark];
}

export type TabButtonProps = uui.TabButtonProps & TabButtonMods;

export const TabButton = withMods<uui.TabButtonProps, TabButtonMods>(uui.TabButton, applyTabButtonMods);
