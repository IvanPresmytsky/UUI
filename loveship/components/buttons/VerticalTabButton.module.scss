@use '../../assets/styles/scss/index' as *;

.root {
    border-bottom: 0 none;
    border-left: 2px solid transparent;
    text-align: left;

    &:global(.uui-active) {
        border-left: 2px solid;
        border-bottom: 0 none;
    }
}
