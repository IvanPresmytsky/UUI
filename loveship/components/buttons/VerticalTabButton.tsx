import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';
import css from './VerticalTabButton.module.scss';

function applyVerticalTabButtonMods() {
    return [css.root];
}

export const VerticalTabButton = createSkinComponent<uui.VerticalTabButtonProps, uui.TabButtonProps>(
    uui.TabButton,
    () => null,
    applyVerticalTabButtonMods,
);
