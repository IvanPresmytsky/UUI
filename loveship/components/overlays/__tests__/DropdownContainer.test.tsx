import React from 'react';
import { renderer } from '@epam/uui-test-utils';
import { DropdownContainer } from '../DropdownContainer';
import { Button } from '../../buttons';

describe('DropdownContainer', () => {
    it('should be rendered correctly', () => {
        const tree = renderer
            .create(
                <DropdownContainer>
                    <Button />
                </DropdownContainer>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly with extra props', () => {
        const tree = renderer
            .create(
                <DropdownContainer color="night700" width={ 200 } height={ 300 } focusLock={ false }>
                    <Button />
                </DropdownContainer>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
