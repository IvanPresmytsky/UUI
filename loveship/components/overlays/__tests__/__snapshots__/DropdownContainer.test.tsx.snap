// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownContainer should be rendered correctly 1`] = `
<div
  className="uui-dropdown-body uui-dropdown_container root uui-color-white -lock-focus container"
  style={
    Object {
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
    }
  }
  tabIndex={0}
>
  <button
    className="container uui-button-box uui-enabled root uui-button uui-fill-solid uui-color-primary size-36 size-36 style-square"
    onClick={[Function]}
    tabIndex={-1}
    type="button"
  />
</div>
`;

exports[`DropdownContainer should be rendered correctly with extra props 1`] = `
<div
  className="uui-dropdown-body uui-dropdown_container root uui-color-night700 -lock-focus container"
  style={
    Object {
      "maxWidth": undefined,
      "minHeight": 300,
      "minWidth": 200,
    }
  }
  tabIndex={0}
>
  <button
    className="container uui-button-box uui-enabled root uui-button uui-fill-solid uui-color-primary size-36 size-36 style-square"
    onClick={[Function]}
    tabIndex={-1}
    type="button"
  />
</div>
`;
